src/build/emitter.ts

```diff
@@ -1786,6 +1788,38 @@ export function emitWebIdl(
     printer.printLine("}");
   }
 
+  function emitES6ToStringTag() {
+    printer.reset();
+    printer.printLine("/////////////////////////////");
+    printer.printLine(`/// ${global} [Symbol.toStringTag] APIs`);
+    printer.printLine("/////////////////////////////");
+
+    const toStringTagProperty = {
+      type: "USVString",
+      name: "[Symbol.toStringTag]",
+      readonly: true,
+      comment:
+        "The well-known symbol @@toStringTag.\n\n" +
+        "[MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag)",
+    };
+
+    const interfaces = getElements(webidl.interfaces, "interface");
+    interfaces.sort(compareName).forEach((i) => {
+      if (!i.noInterfaceObject && !i.namespace && !i.extends && !i.implements) {
```

Comment:
I get `i.extends` as it'll be inherited from parent interface, but why others checks? We do need to filter out i.legacyNamespace and emit it inside a namespace, see also:

https://github.com/microsoft/TypeScript-DOM-lib-generator/blob/6621974de7d3d03ade59760ea4b1cdb12698f492/src/build/emitter.ts#L1491